semtypes/src/main/java/io/ballerina/semtype/Common.java

```diff
@@ -114,22 +116,46 @@
         return Conjunction.and(atom, next);
     }
 
-    public static List<SemType> shallowCopyTypes(SemType[] v) {
-        return Arrays.asList(v);
+    public static SemType[] shallowCopyTypes(SemType[] v) {
+        return Arrays.copyOf(v, v.length);
     }
 
     public static List<SemType> shallowCopyTypes(List<SemType> v) {
         return new ArrayList<>(v);
     }
 
-    public static String[] shallowCopyTypes(String[] v) {
+    public static String[] shallowCopyStrings(String[] v) {
         return Arrays.copyOf(v, v.length);
     }
 
     public static boolean notIsEmpty(TypeCheckContext tc, SubtypeData d) {
         return false;
     }
 
+    // Returns whether s1.codePoints < s2.codePoints
+    public static boolean codePointCompare(String s1, String s2) {
```

Comment:
Please find the string ordering requirements [here](https://ballerina.io/spec/lang/master/#ordering)